scripts/Muon/GUI/ElementalAnalysis2/grouping_widget/ea_grouping_tab_model.py

```diff
@@ -0,0 +1,91 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from Muon.GUI.ElementalAnalysis2.ea_group import EAGroup
+from enum import Enum
+
+
+class RowValid(Enum):
+    invalid_for_all_runs = 0
+    valid_for_all_runs = 2
+    valid_for_some_runs = 1
+
+
+class EAGroupingTabModel(object):
```

Comment:
This class is nice a neat so good job. We can remove the inheritance of `object` as this would only be required if we still supported python 2